### DISCUSSION TITLE
Allow to set metric timestamp from query results

### DISCUSSION THREAD

**Discussion by Koaxiel** [burningalchemist/sql_exporter#301] [Ideas]
Hello,

I would like to know your opinion on the addition of a feature to set the metric' timestamp according to the result of a query.
Indeed, the [Prometheus format](https://prometheus.io/docs/instrumenting/exposition_formats/) allows us to add the timestamp in addition to the value of a metric.
From the Golang client, the [NewMetricWithTimestamp](https://github.com/prometheus/client_golang/blob/v1.16.0/prometheus/metric.go#L146) function allows to do that easily.
We can imagine having a new setting (like key_timestamp) to define the column to use to set the metric' timestamp.

I know it's not a common need, but it's useful when using DBs like Vertica to make heavy analyses and to be able to keep the initial timestamp (the one when the DB is injected) in the results.

WDYT?

**Comment by burningalchemist** [comment]
Hey @Koaxiel, thank you for your suggestion! I think that having timestamps in `sql_exporter` might come in handy.

My current concern is what shall be picked as a timestamp. It'd be cool if we could list possible scenarios (your input is appreciated) to understand how we can offer a flexible yet more/less reliable interface (e.g. `key_timestamp` to pick the field - which I like - or some other setting).

As for ETAs, I'm currently through process of covering code with tests and a few job-related (multiple connections) enhancements. I think code really needs some refactoring, but without properly written tests we're at risk of breaking current things. On the client side it's indeed the function you've mentioned, but it might be a little be complicated in this implementation.

Anyways, I like the idea, and I'll try to prototype it sooner. 👍 If it goes faster than expected, I'll share the results and a branch to test, if you're up to it. 🙂

**Comment by burningalchemist** [comment]
It's resolved in #429
